### ISSUE TITLE
Linking fails on Debian 11 (Stable)

### ISSUE THREAD

**Issue by vanfanel** [ScrelliCopter/gish#1]
Hi there,

I tried to build this on Debian 11 Stable, but upon linking I get this:
```

[100%] Linking C executable gish
/usr/bin/ld: CMakeFiles/gish.dir/src/game/custom.c.o:(.bss+0x9da0): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/custom.c.o:(.bss+0x9da4): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/damage.c.o:(.bss+0x0): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/damage.c.o:(.bss+0x4): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/editor.c.o:(.bss+0x10060): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/editor.c.o:(.bss+0x10064): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/game.c.o:(.bss+0x25c): multiple definition of `numofsprites'; CMakeFiles/gish.dir/src/game/animation.c.o:(.bss+0x1204): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/game.c.o:(.bss+0x260): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/game.c.o:(.bss+0x264): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/level.c.o:(.bss+0x507080): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/level.c.o:(.bss+0x507084): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/lighting.c.o:(.bss+0x0): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/lighting.c.o:(.bss+0x4): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/logic.c.o:(.bss+0x0): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/logic.c.o:(.bss+0x4): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/mappack.c.o:(.bss+0x0): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/mappack.c.o:(.bss+0x4): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/object.c.o:(.bss+0x180424): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/object.c.o:(.bss+0x180428): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/objedit.c.o:(.bss+0x0): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/objedit.c.o:(.bss+0x4): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/objfunc.c.o:(.bss+0x0): multiple definition of `numofsprites'; CMakeFiles/gish.dir/src/game/animation.c.o:(.bss+0x1204): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/objfunc.c.o:(.bss+0x4): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/objfunc.c.o:(.bss+0x8): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/physics.c.o:(.bss+0xf5024): multiple definition of `numofsprites'; CMakeFiles/gish.dir/src/game/animation.c.o:(.bss+0x1204): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/physics.c.o:(.bss+0xf5028): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/physics.c.o:(.bss+0xf502c): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/player.c.o:(.bss+0x784): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/player.c.o:(.bss+0x788): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/render.c.o:(.bss+0x1cc): multiple definition of `numofsprites'; CMakeFiles/gish.dir/src/game/animation.c.o:(.bss+0x1204): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/render.c.o:(.bss+0x1c4): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/render.c.o:(.bss+0x1c8): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/ropeedit.c.o:(.bss+0x28010): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/ropeedit.c.o:(.bss+0x28014): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/setup.c.o:(.bss+0x0): multiple definition of `numofsprites'; CMakeFiles/gish.dir/src/game/animation.c.o:(.bss+0x1204): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/setup.c.o:(.bss+0x4): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/setup.c.o:(.bss+0x8): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/sprite.c.o:(.bss+0x6800): multiple definition of `numofsprites'; CMakeFiles/gish.dir/src/game/animation.c.o:(.bss+0x1204): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/texture.c.o:(.bss+0x30): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/texture.c.o:(.bss+0x34): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/vsmode.c.o:(.bss+0x0): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/game/vsmode.c.o:(.bss+0x4): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/physics/object.c.o:(.bss+0x0): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/physics/object.c.o:(.bss+0x4): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/physics/particle.c.o:(.bss+0x220004): multiple definition of `levelfriction'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/gish.dir/src/physics/particle.c.o:(.bss+0x220008): multiple definition of `levelblocknum'; CMakeFiles/gish.dir/src/game/ai.c.o:(.bss+0x4): first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/gish.dir/build.make:963: gish] Error 1
make[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/gish.dir/all] Error 2
make: *** [Makefile:103: all] Error 2
```

Maybe the order of the source files is not right or something like that?

Thanks!

**Comment by vanfanel** [comment]
@ScrelliCopter The problem has a very easy solution: pass `-fcommon` to the CFLAGS and linking works.
As noted here:
https://github.com/EXL/Gish/issues/7

**Comment by ScrelliCopter** [comment]
Fixed in [beb4fd6](https://github.com/ScrelliCopter/gish/commit/beb4fd689ac3a290ef48d5faef9ab9a816367f5c).
